pkg/multiclusterengine/olm/v1/clustercatalog.go

```
@@ -0,0 +1,168 @@
+	// Skip TLS verification for in-cluster service communication
+	// #nosec G402 -- catalogd service uses self-signed cert for in-cluster communication
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+	client := &http.Client{Transport: tr}
+
+	resp, err := client.Get(url)
+	if err != nil {
+		return false, fmt.Errorf("failed to query catalogd API: %w", err)
+	}
+	defer resp.Body.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `resp.Body.Close` is not checked

(errcheck)